cookbooks/node/recipes/yarn.rb

```diff
@@ -0,0 +1,50 @@
+# Installs latest Yarn and sets up links properly
+# TODO:  Execute the recipe only if a newer version of yarn is available
+# (checking by hitting https://github.com/yarnpkg/yarn/releases/latest and grabbing the redirection
+
+yarn_download_url = "https://yarnpkg.com/latest.tar.gz"
+
+directory "/tmp/yarn" do
+  recursive true
+  action :delete
+end
+
+directory "/tmp/yarn" do
+  mode 0755
+  action :create
+end
+
+remote_file "/tmp/yarn/yarn-latest.tar.gz" do
+  source "#{yarn_download_url}"
+  mode 0644
+  backup 0
+end
+
+execute "unarchive Yarn" do
+  cwd "/tmp/yarn"
+  command "tar zxf yarn-latest.tar.gz"
+end
+
```

Comment:
Maybe add a sleep here to allow the tar to complete before proceeding to the next step?